azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -36,14 +36,15 @@ class AbstractIoTHubClient(object):
     This class needs to be extended for specific clients.
     """
 
-    def __init__(self, iothub_pipeline):
+    def __init__(self, iothub_pipeline, upload_pipeline):
         """Initializer for a generic client.
 
         :param iothub_pipeline: The pipeline used to connect to the IoTHub endpoint.
         :type iothub_pipeline: :class:`azure.iot.device.iothub.pipeline.IoTHubPipeline`
         """
         self._iothub_pipeline = iothub_pipeline
-        self._edge_pipeline = None
+        self._upload_pipeline = upload_pipeline
```

Comment:
>upload_pipeline [](start = 32, length = 15)

Do we need to have upload_pipeline and _invoke_method_pipeline as 2 distinct things?  Could there be a single pipeline that implements both?  I think that would save a ton of code and I don't see why they _need_ to be separate.